app/controllers/api/v1/protocol_controller.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class ProtocolController < ApiController
+      include ::Concerns::IsJwtAuthenticated
```

Comment:
Is er een reden dat we niet in alle controllers die van ApiController deriven overal IsLoggedInWithAnyMethod gebruiken? ipv soms IsJwtAuthenticated zoals hier en soms IsLoggedInWithAnyMethod?

Ik zal het je nog sterker vertellen. Volgens mij willen we in alle ApiControllers dat mensen expliciet via Jwt zijn ingelogd. Dus overal IsJwtAuthenticated ipv IsLoggedInWithAnyMethod. Dan hebben we een mooie scheiding tussen ApiController backend methoden die je alleen aan kan roepen met een JWT inlog, ApiTokenController backend methoden die je alleen via basicAuth, en front-end controllers, die je kan aanroepen via jwt token of "het klikken op een vragenlijst link" (wat we in de code verwarrend genoeg TokenAuthenticator noemen).

Dus ik stel iets voor als:
**ApiController** -> doet niks behalve het definieeren van wat common methoden bij het gooien van errors ofzo
**JWTApiController < ApiController** -> Dit wordt de nieuwe base class voor de api controllers protocol subscriptions controller, protocol controller, person controller ed. En hier staat dus in include ::Concerns::IsJwtAuthenticated, zodat we dat niet steeds hoeven te herhalen (en mogelijk vergeten) in alle deriving controllers.
**ApiTokenApiController < ApiController** -> Include standaard basic auth, en is de base van die andere protocol subscriptions controller waarmee je protocol subscriptions kan starten voor andere personen.

En dan gebruiken we die andere inlog (dwz dat je op een vragenlijst link klikt), dat we die inlog alleen gebruiken voor front-end controllers (questionnaire controller, reward controller enz, die we gebruiken om vragenlijsten en UI te laten zien). En daar (en alleen daar) gebruiken we dan IsloggedInWithAnyMethod (want dan kunnen we mensen vanuit het base-platform backend mooi met een linkje naar https://sdv-questionnaires.researchable.nl redirecten zonder er iets achter te hoeven zetten (wel met een auth/token param), en dan worden ze geredirect naar de eerste vragenlijst die voor hun klaarstaat. Wil je het jezelf makkelijk maken dan gooi je ook hier weer een controller tussen (bijv LoggedInController < ApplicationController, die dan include isLoggedInWithAnyMethod doet, en waar de questionnaire controller, rewards controller e.d. van erven zodat we daar alleen kunnen komen als we ingelogd zijn).

Het is namelijk niet mooi en erg verwarrend dat we nu overal die verschillende concerns moeten includen in alle deriving controllers, grote kans dat we dat ergens vergeten en dan ligt de hele api open. Ook is het zo beter gestructureerd (dat je bijv met een gewone "klik op vragenlijst inlog" niet zomaar bij de API controllers mag komen).